src/containers/Signup/index.jsx

```diff
@@ -0,0 +1,216 @@
+import Loader from "../AmazonMusic/components/Loader";
+import { SIDE_CONTAINER_DISPLAY } from "../AmazonMusic/constants";
+import { Box, Divider, IconButton, Stack, TextField } from "@mui/material";
+import ArrowBackIosNewIcon from "@mui/icons-material/ArrowBackIosNew";
+
+const INITIAL_STATE = {
+  name: "",
+  email: "",
+  password: "",
+};
+
+const INITIAL_ERROR_DATA = {
+  nameError: "",
+  emailError: "",
+  passwordError: "",
+};
+
+function Signup() {
+  const [userData, setUserData] = useState(INITIAL_STATE);
+  const [errorData, setErrorData] = useState(INITIAL_ERROR_DATA);
+
+  const { loading, msgDisplayed } = useSelector(
+    (state) => state.registeredUser
+  );
+  const dispatch = useDispatch();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (msgDisplayed) {
+      routeTologin();
+    }
+  }, [msgDisplayed]);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setUserData({ ...userData, [name]: value });
+  };
+
+  const register = (event) => {
+    event.preventDefault();
+    setErrorData((obj) => ({ ...obj, ...INITIAL_ERROR_DATA }));
+    console.log("clicked register");
+    const errorResult = validation(userData);
+
+    if (errorResult) {
+      setErrorData((obj) => ({ ...obj, ...errorResult }));
+      return;
+    } else {
+      //dispatch signup action
+      //submit userData in Redux
+      console.log(userData);
+      dispatch(signup({ ...userData, appType: "music" }));
+      setUserData((obj) => ({ ...obj, ...INITIAL_STATE }));
+      setErrorData((obj) => ({ ...obj, ...INITIAL_ERROR_DATA }));
+    }
+  };
+
+  const validation = (userData) => {
+    const { name, email, password } = userData;
+
+    if (!name || !email || !password) {
+      return {
+        nameError: "Please enter all fields",
+        emailError: "Please enter all fields",
+        passwordError: "Please enter all fields",
+      };
+    }
+    let nameError =
+      name.length > 3 ? "" : "name must contain atleast 3 letters";
+    let emailError =
+      email.includes("") && email.split("@")[0].length > 3
+        ? ""
+        : "Please enter valid email";
+    let passwordError =
+      password.length > 3 ? "" : "password must contain atleast 3 letters";
+    console.log(nameError, emailError, passwordError);
+
+    if (!emailError && !passwordError && !nameError) return false;
+    return {
+      nameError,
+      emailError,
+      passwordError,
+    };
+  };
+
+  const routeTologin = () => {
+    navigate(LINKS.login);
+  };
+
+  const { name, email, password } = userData;
+  const { nameError, emailError, passwordError } = errorData;
+
+  if (loading) return <Loader />;
+
+  return (
+    <Stack
+      sx={{
+        display: "flex",
+        flexDirection: "row",
+        justifyContent: "space-between",
+        minHeight: "100vh",
+        backgroundColor: "rgba(255, 255, 255)",
+        color: "hsl(0, 0%, 4%)",
+        width: "100dvw",
+      }}
```

Comment:
create a JS file inside that create a object and write this css there and call that object here.

Reply:
file name can be : index.style.js